rewrite-maven/src/main/java/org/openrewrite/maven/AddManagedDependencyVisitor.java

```diff
@@ -0,0 +1,175 @@
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.maven;
+
+import lombok.RequiredArgsConstructor;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Validated;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.maven.internal.InsertDependencyComparator;
+import org.openrewrite.maven.tree.MavenMetadata;
+import org.openrewrite.semver.LatestRelease;
+import org.openrewrite.semver.Semver;
+import org.openrewrite.semver.VersionComparator;
+import org.openrewrite.xml.AddToTagVisitor;
+import org.openrewrite.xml.XPathMatcher;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.Objects;
+
+import static java.util.Collections.emptyList;
+
+@RequiredArgsConstructor
+public class AddManagedDependencyVisitor extends MavenIsoVisitor<ExecutionContext> {
+    private static final XPathMatcher MANAGED_DEPENDENCIES_MATCHER = new XPathMatcher("/project/dependencyManagement/dependencies");
+
+    private final String groupId;
+    private final String artifactId;
+    private final String version;
+
+    @Nullable
+    private final String versionPattern;
+
+    @Nullable
+    private final String scope;
+
+    @Nullable
+    private final Boolean releasesOnly;
+
+    @Nullable
+    private final String type;
+
+    @Nullable
+    private final String classifier;
+
+    @Nullable
+    private VersionComparator versionComparator;
+
+    @Nullable
+    private String resolvedVersion;
+
+    @Override
+    public Xml.Document visitDocument(Xml.Document document, ExecutionContext ctx) {
+
+        Validated versionValidation = Semver.validate(version, versionPattern);
+        if (versionValidation.isValid()) {
+            versionComparator = versionValidation.getValue();
+        }
+
+        String versionToUse = findVersionToUse(ctx);
+        if (versionToUse.equals(existingManagedDependencyVersion())) {
+            return document;
+        }
+
+        Xml.Document doc = super.visitDocument(document, ctx);
+        Xml.Tag root = document.getRoot();
+        if (root.getContent() != null) {
+            Xml.Tag dependencyManagementTag = root.getChild("dependencyManagement").orElse(null);
+            if (dependencyManagementTag == null) {
+                doc = (Xml.Document) new AddToTagVisitor<>(root, Xml.Tag.build("<dependencyManagement>\n<dependencies/>\n</dependencyManagement>"),
+                        new MavenTagInsertionComparator(root.getContent())).visitNonNull(doc, ctx);
+            } else if (dependencyManagementTag.getContent() == null || dependencyManagementTag.getContent().isEmpty()) {
```

Comment:
I think we might come across dependency management tags that only contain comments.
What do you think about using `getChildren()` to return the tags instead of `getContent()`?

Reply:
Thanks @traceyyoshima that's good to know.  
In this case, the null check is for the comparators used for adding new tags.. 
`new MavenTagInsertionComparator(root.getContent())).visitNonNull(doc, ctx);`

Reply:
@pway99 Ah I see, so `dependencyManagementTag.getContent().isEmpty()` isn't meant to check if there aren't any tags?

Reply:
actually misinterpreted your comment, now I see your concern for a this statement `dependencyManagementTag.getContent().isEmpty()`